zerver/views/streams.py

```diff
@@ -318,6 +318,13 @@ def update_stream_backend(
         if is_private and stream.id in default_stream_ids:
             raise JsonableError(_("Default streams cannot be made private."))
 
+        if (
+            not is_private
+            and history_public_to_subscribers is False
+            and not stream.realm.is_zephyr_mirror_realm
+        ):
+            raise JsonableError(_("Invalid parameters"))
```

Comment:
nit: the commit message is quite wide; 84 columns, I think. [Our guidelines say](https://zulip.readthedocs.io/en/latest/contributing/version-control.html#message-body):
> Any paragraph content in the commit message should be line-wrapped to about 68 characters per line, but no more than 70, so that your commit message will be reasonably readable in `git log` in a normal terminal.

Please wrap the messages to keep them readable. :slightly_smiling_face: See the suggestions at that link for setting up Git and your editor to make this automatic.

Same goes for the other commits in the branch, though less extreme.

To illustrate, here's what this commit looks like for me:
![image](https://user-images.githubusercontent.com/28173/171301849-d23fb20f-f2fd-4d4f-9666-d2745f986c9c.png)